DriveKitCommonUI/Utils/DKScoreTypeLevel.swift

```diff
@@ -58,6 +58,29 @@ public enum DKScoreTypeLevel: CaseIterable {
         }
         // swiftlint:enable no_magic_numbers
     }
+
+    public static func getLevel(for score: Double, scoreType: DKScoreType) -> DKScoreTypeLevel {
+        let steps = scoreType.getSteps()
+        let expectedLevelsCount = 8
+        guard steps.count == expectedLevelsCount else {
+            return .veryBad
+        }
+        if score < steps[1] {
+            return .veryBad
+        } else if score < steps[2] {
+            return .bad
+        } else if score < steps[3] {
+            return .notGood
+        } else if score < steps[4] {
+            return .medium
+        } else if score < steps[5] {
+            return .great
+        } else if score < steps[6] {
+            return .veryGood
+        } else {
+            return .excellent
+        }
+    }
```

Comment:
```suggestion
    public static func getLevel(for score: Double, scoreType: DKScoreType) -> DKScoreTypeLevel {
        guard
            let level = Self.allCases.filter({
                $0.scoreLevels(for: scoreType).contains(score)
            }).first
        else {
            if score < Self.veryBad.scoreLevels(for: scoreType).lowerBound {
                return .veryBad
            } else {
                return .excellent
            }
        }
        
        return level
    }
```

On peut réutiliser la méthode `scoreLevels(for:)`, ce qui évite de remettre des warnings `no_magic_numbers` et de dupliquer d'une certaine façon le code relatif au steps. Qu'en penses-tu ?

Reply:
corrigé dans e2e52dd4cf71b9d46fcbe538640d7ae2fdc487eb